sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -189,21 +202,19 @@ public Builder withValidation() {
   /** A fluent {@link PipelineOptions} builder. */
   public static class Builder {
     private final String defaultAppName;
-    private final String[] args;
-    private final Map<String, String> options;
+    private final Collection<Function<Builder, Multimap<String, String>>> options;
     private final boolean validation;
     private final boolean strictParsing;
     private final boolean isCli;
 
     // Do not allow direct instantiation
     private Builder() {
-      this(null, null, false, true, false);
+      this(new ArrayList<>(), false, true, false);
     }
 
-    private Builder(String[] args, Map<String, String> options, boolean validation,
-        boolean strictParsing, boolean isCli) {
+    private Builder(Collection<Function<Builder, Multimap<String, String>>> options,
```

Comment:
`Properties` isn't a multimap, actually. Entries overwrite each other when they are specified more than once.

Reply:
Hmm, thought https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java#L1546 concatenate it when parsing the args and values are differents (means --foo=a --foo=a would loose a value compared to --foo=a --foo=b).

Reply:
Ah, makes sense. So cmdline is multimap while Properties are limited to just a map. SGTM

Reply:
TBH it is not a good cmdline interface to do this

Reply:
well tempted to say +-0, I saw so much times the debate = vs space to delimit the key/value and single option with N values vs N times the options that I just pick one now and move forward ;).